src/shared/clients/lambdaClient.ts

```diff
@@ -74,6 +74,33 @@ export class DefaultLambdaClient {
         }
     }
 
+    public async getFunctionUrlConfigs(name: string): Promise<Lambda.FunctionUrlConfigList> {
+        getLogger().debug(`GetFunctionUrlConfig called for function: ${name}`)
+        const client = await this.createSdkClient()
+
+        try {
+            const request = client.listFunctionUrlConfigs({ FunctionName: name })
+            const response = await request.promise()
+            // prune `Code` from logs so we don't reveal a signed link to customer resources.
+            getLogger().debug('GetFunctionUrlConfig returned response (code section pruned): %O', {
+                ...response,
+                Code: 'Pruned',
+            })
+            return response.FunctionUrlConfigs
+        } catch (e) {
+            getLogger().error('Failed to get function url config: %s', e)
+            throw e
+        }
+    }
+
+    public async getFunctionUrl(name: string): Promise<string | void> {
```

Comment:
We could remove this method and instead put the logic in `copyLambdaUrl.ts` directly. The less public methods the client wrapper has, the better.

Reply:
> The less public methods the client wrapper has, the better.

why?

Reply:
I agree Jaden, I've removed this function and instead find and grab the function URL in the business logic.